frame/staking/src/benchmarking.rs

```diff
@@ -477,6 +405,156 @@ benchmarks! {
 			call.dispatch(RawOrigin::Signed(caller.clone()).into())?;
 		}
 	}
+
+	submit_solution_initial {
+		// NOTE: this benchmark requires: `v > w`, `n > a` and `a > w`. Talk to @kianenigma if you
+		// want to know why.
+		// number of validator intentions
+		let v in 50 .. 100;
+		// number of nominators
+		let n in 200 .. 500;
+		// number of winners.
+		let w in 10 .. 40;
+		// number of assignments. Basically, number of active nominators.
+		let a in 50 .. 200;
+		// NOTE: we could make the edge per assignment also variable, but 1. unlikely to bring much
+		// accuracy change 2. makes the setup of the bench a few orders of magnitude more annoying.
+
+		MinimumValidatorCount::put(0);
+		create_validators_with_nominators_for_era::<T>(v, n, MAX_NOMINATIONS, false)?;
+
+		// override the number of winners that we desire.
+		ValidatorCount::put(w);
+
+		// needed for the solution to be generates.
+		assert!(<Staking<T>>::create_stakers_snapshot().0);
+		let (winners, compact, score, size) = get_seq_phragmen_solution::<T>(true);
+
+		// trim some edges.
+		let (new_compact, new_score) = trim_compact_to_assignments::<T>(
+			compact,
+			winners.clone(),
+			a as usize,
+		)?;
+
+		// needed for the solution to be accepted
+		<EraElectionStatus<T>>::put(ElectionStatus::Open(T::BlockNumber::from(1u32)));
+
+		let era = <Staking<T>>::current_era().unwrap_or(0);
+		let caller: T::AccountId = account("caller", n, SEED);
+	}: {
+		let result = <Staking<T>>::submit_election_solution(
+			RawOrigin::Signed(caller.clone()).into(),
+			winners,
+			new_compact,
+			new_score.clone(),
+			era,
+			size,
+		);
+		assert_eq!(result, Ok(().into()));
+	}
+	verify {
+		// new solution has been accepted.
+		assert_eq!(<Staking<T>>::queued_score().unwrap(), new_score);
+	}
+
+	submit_solution_weaker {
+		// TODO: whatever this weight ends up being, so how it will pan our for a solution which has
+		// 1000 validators, and like 100,000 nominators
```

Comment:
Don't understand this comment, can you correct or elaborate?
Do you mean
"Whatever the weight of `submit_solution` ends up being we want to know how it will pan out for a solution with 1000 validators and 100_000 nominators"
?

Reply:
So I am worried that the weight formula, once applied to a large network size, ends up being something like 500 times the block weight limit :D